databricks_cli/stack/api.py

```diff
@@ -0,0 +1,338 @@
+RESOURCE_PROPERTIES = 'properties'
+
+# Deployed Resource Fields
+RESOURCE_PHYSICAL_ID = 'physical_id'
+RESOURCE_DEPLOY_OUTPUT = 'deploy_output'
+RESOURCE_DEPLOY_TIMESTAMP = 'timestamp'
+CLI_VERSION_KEY = 'cli_version'
+
+
+class StackApi(object):
+    def __init__(self, api_client):
+        self.jobs_client = JobsApi(api_client)
+
+    def _load_json(self, path):
+        """
+        Parse a json file to a readable dict format.
+        Returns an empty dictionary if the path doesn't exist.
+
+        :param path: File path of the JSON stack configuration template.
+        :return: dict of parsed JSON stack config template.
+        """
+        stack_conf = {}
+        if os.path.exists(path):
+            with open(path, 'r') as f:
+                stack_conf = json.load(f)
+
+        return stack_conf
+
+    def _json_type_handler(self, obj):
+        """
+        Helper function to convert certain objects into a compatible JSON format.
+
+        Right now, converts a datetime object to an integer timestamp.
+
+        :param obj: Object that may be a datetime object.
+        :return: Timestamp integer if object is a datetime object.
+        """
+        if isinstance(obj, datetime):
+            # Get timestamp of datetime object- works with python2 and 3
+            return int(time.mktime(obj.timetuple()))
+        raise TypeError("Object of type '%s' is not JSON serializable" % type(obj))
+
+    def _save_json(self, path, data):
+        """
+        Writes data to a JSON file.
+
+        :param path: Path of JSON file.
+        :param data: dict- data that wants to by written to JSON file
+        :return: None
+        """
+        with open(path, 'w+') as f:
+            json.dump(data, f, indent=2, sort_keys=True, default=self._json_type_handler)
+
+    def _generate_stack_status_path(self, stack_path):
+        """
+        Given a path to the stack configuration template JSON file, generates a path to where the
+        deployment status JSON will be stored after successful deployment of the stack.
+
+        :param stack_path: Path to the stack config template JSON file
+        :return: The path to the stack status file.
+
+        >>> self._generate_stack_status_path('./stack.json')
+        './stack.deployed.json'
+        """
+        stack_path_split = stack_path.split('.')
+        stack_path_split.insert(-1, STACK_STATUS_INSERT)
+        return '.'.join(stack_path_split)
+
+    def _get_resource_to_status_map(self, stack_status):
+        """
+        Returns a dictionary that maps a resource's (id, service) to the resource's status
+        from the last deployment
+
+        The key for this dictionary is the resource's (id, service) so that we don't load
+        persisted resources with the wrong resource service.
+        """
+        return {(resource_status[RESOURCE_ID], resource_status[RESOURCE_SERVICE]): resource_status
+                for resource_status in stack_status[STACK_DEPLOYED]}
```

Comment:
This one line function is only used once and it is hard to understand it without the context on how it was used? How about remove this thin function and just move this line into `deploy_stack` directly? The python codes itself is self explained.